ts/features/bonus/bpd/saga/networking/ranking.ts

```diff
@@ -63,3 +80,113 @@ export function* bpdLoadRaking(
     return left<Error, ReadonlyArray<BpdRankingReady>>(getError(e));
   }
 }
+
+// convert the network payload ranking into the relative app domain model
+const convertRankingArrayV2 = (
+  rankings: CitizenRankingMilestoneResourceArray
+): ReadonlyArray<BpdRankingReady> =>
+  rankings.map<BpdRankingReady>(rr => ({
+    ...rr,
+    awardPeriodId: rr.awardPeriodId as AwardPeriodId,
+    pivot: extractPivot(rr.milestoneResource),
+    kind: "ready"
+  }));
+
+const extractPivot = (
+  mr: MilestoneResource | undefined
+): BpdPivotTransaction | undefined => {
+  if (mr?.idTrxPivot !== undefined && mr?.cashbackNorm !== undefined) {
+    return {
+      amount: mr.cashbackNorm,
+      idTrx: mr.idTrxPivot as BpdTransactionId
+    };
+  }
+  return undefined;
+};
+
+/**
+ * Load the ranking for all the periods
+ * @param getRanking
+ */
+export function* bpdLoadRakingV2(
+  getRanking: ReturnType<typeof BackendBpdClient>["getRankingV2"]
+): Generator<Effect, Either<Error, ReadonlyArray<BpdRankingReady>>, any> {
+  try {
+    void mixpanelTrack(mixpanelActionRequest);
+    const getRankingResult: SagaCallReturnType<typeof getRanking> = yield call(
+      getRanking,
+      {} as any
+    );
+    if (getRankingResult.isRight()) {
+      if (getRankingResult.value.status === 200) {
+        void mixpanelTrack(mixpanelActionSuccess, {
+          count: getRankingResult.value.value?.length
+        });
+        return right<Error, ReadonlyArray<BpdRankingReady>>(
+          convertRankingArrayV2(getRankingResult.value.value)
+        );
+      } else if (getRankingResult.value.status === 404) {
+        void mixpanelTrack(mixpanelActionSuccess, {
+          count: 0
+        });
+        return right<Error, ReadonlyArray<BpdRankingReady>>([]);
+      } else {
+        return left<Error, ReadonlyArray<BpdRankingReady>>(
+          new Error(`response status ${getRankingResult.value.status}`)
+        );
+      }
+    } else {
+      return left<Error, ReadonlyArray<BpdRankingReady>>(
+        new Error(readableReport(getRankingResult.value))
+      );
+    }
+  } catch (e) {
+    void mixpanelTrack(mixpanelActionFailure, {
+      reason: getError(e).message
```

Comment:
```suggestion
      reason: getNetworkError(e).message
```
Could it be better to have a readable message of timeout errors?

Reply:
Yes, totally better, I've changed it in `getNetworkErrorMessage(e)`a69a32d0773d6db8dace628535c3f774e8715324